### DISCUSSION TITLE
Questions about train and validation

### DISCUSSION THREAD

**Discussion by kennedyk1** [ultralytics/yolov5#13394] [General]
Hello everyone!

I have a few questions about the use of train, test, and validation sets during my model training process that remain unclear. I’ll describe my situation below to provide more context:

- I split my dataset into two folders: `train` and `test`.
- In the `rgb.yaml` file, I set the test folder as the `validation` set.

```yaml
path: '/home/user/my_datasets/'
train: 'dataset_1/train/rgb/'
val: 'dataset_1/test/rgb/'
# class names
names:
  0: 'human'
```

- In the main code, I disabled early stopping by setting `patience=0` and trained the model for 50 epochs.
- I used the `low-scratch.yaml` hyperparameter profile to enhance data augmentation.

My main code:
```python
from ultralytics import YOLO
datasets = ['rgb.yaml']
yolo_models = ['yolov6s.yaml','yolov5s.yaml','yolov8s.yaml','yolov9s.yaml','yolov10s.yaml','yolo11s.yaml']
epochs = 50

for dataset in datasets:
    for yolo_model in yolo_models:
        model = YOLO(yolo_model,task='detect') # Create a new model to train
        model.train(
            epochs = epochs, #NUMBER EPOCHS TO TRAIN
            batch = 0.9, #TO USE 90% GPU MEMORY
            single_cls = True, #TO TRAIN WITH ONLY ONE CLASS
            imgsz = 640, # TO REZISE IMAGES, DEFAULT 640
            data = dataset, #TO DEFINE YAML DATASET FILE
            patience = 0, #TO AVOID EARLY STOP
            save = True, #TO SAVE CHECKPOINTS AND FINAL MODEL WEIGHTS
            project = yolo_model.split('.')[0], #NAME OF PROJECT
            name = dataset.split('.')[0], #SUB-NAME OF PROJECT or MODALITY
            plots = True #TO SHOW PLOTS OF TRAINNING AND VALIDATION METRICS
        )
```

Based on the details above, I have the following questions:

-    When is data augmentation applied?
-   Is data augmentation performed on the train set, the validation set, or both?
-    At what stage is validation conducted?
-    Can my `test` (used as validation) folder be considered "unseen" data, given that I have disabled early stopping?
-    Can I consider given metrics (P, R, mAP50 and mAP50:95) at end training as test evaluation, given that I have disabled early stopping?
